src/compose/service.ts

```diff
@@ -528,10 +528,30 @@ export class Service {
 		}
 		expose = _.uniq(expose);
 
-		const tmpfs: string[] = [];
-		_.each((container.HostConfig as any).Tmpfs, (_v, key) => {
-			tmpfs.push(key);
-		});
+		const tmpfs: string[] = _.uniq(
+			([] as string[]).concat(
+				Object.keys(container.HostConfig?.Tmpfs || {}),
+				// also consider mount api
+				(container.HostConfig?.Mounts || [])
+					.filter(({ Type }) => Type === 'tmpfs')
+					.map(({ Target }) => Target),
+			),
+		);
+
+		// bind and volume mounts
+		const mounts: string[] = _.uniq(
+			([] as string[]).concat(
+				Object.keys(container.Config?.Volumes || {}),
+				container.HostConfig?.Binds || [],
```

Comment:
Removing this line may force recreate of container

Reply:
I removed this line in https://github.com/balena-os/balena-supervisor/commit/3a315db354c2c937b2a3a8d37d3fe00196b6a192 and updated some test data files as a result. Unit tests pass, on-device tests pending.

Reply:
Added this line back with https://github.com/balena-os/balena-supervisor/commit/c8f2c21e84971bd893a273578f4f81683f23386c, since update lock bind mounts are configured as HostConfig.Binds.